File: python/equacaoSegundoGrau.py
# Início do algoritmo

# Importação da biblioteca CMath
from cmath import sqrt

# Declaração de Variáveis
a = 0
b = 0
c = 0
delta = 0
x1 = round(0)
x2 = round(0)

# Inputs que pergunta ao usuário quais números ele irá utilizar para o cálculo.
a = float(input("Digite o número A: "))
b = float(input("Digite o número B: "))
c = float(input("Digite o número C: "))

# A realização da equação de segundo grau.
delta = float(  b** - (4 * a * c))

# Verifica se delta é menor do que zero.
# Else realiza as raizes.
if(delta < 0):
    print("Não existe raizes possíveis.")
else:
    x1 = -b + sqrt(delta) / (2 * a)
    x2 = -b - sqrt(delta) / (2 * a)
    print(f"A raiz 1 é = {x1}, \n A raiz 2 é = {x2}" )

#Fim do algoritmo.

File: ruby/equacaoSegundoGrau.rb
#Início do algoritmo

# Declaração de variáveis
=begin
a = REAL
b = REAL 
c = REAL
delta = REAL 
x1 = REAL 
x2 = REAL
=end

# Importação da biblioteca cmath
require 'cmath' 

# Aqui será pego todos os números digitados pelo o usuário.
puts('Digite o número de A: ')
a = Float(gets)
puts('Digite o número de B: ')
b = Float(gets)
puts('Digite o número de C: ')
c = Float(gets)

# Aqui será feito o calculo para achar o DELTA.
delta = b** -(4 * a * c)

# Verificação caso DELTA for menor do que 0, exibir ao usuário que não possui raizes reais. 
if delta < 0
    puts("Não possui raizes reais.")
else
    # Aqui é feito o calculo final de Bhaskara, fazendo em si a divisão e a multiplicação para achar a raiz quadrada.
    x1 = (-b + CMath.sqrt(delta)) / (2 * a)
    x2 = (-b - CMath.sqrt(delta)) / (2 * a)
    # O final de tudo, aqui mostra ao usuário à raiz de A e a raiz de B.
    puts("""
        A raiz 1 é = #{x1},
        A raiz 2 é = #{x2}
    """) 
end

# Fim do Algoritmo 